nibabel/externals/netcdf.py

```diff
@@ -607,7 +607,10 @@ def _read_var_array(self):
                     pos = self.fp.tell()
                     self.fp.seek(begin_)
                     data = fromstring(self.fp.read(a_size), dtype=dtype_)
-                    data.shape = shape
+                    # Avoid exception for empty variables.
+                    if len(data.shape) != 1 or data.shape[0] != 0:
+                        data.shape = shape
```

Comment:
This is equivalent to:

```Python
if data.shape != (0,):
    data.shape = shape
````

Are there other cases to check against?

Looking at the surrounding code, I *think* this is doing an awful lot to read zero bytes. Maybe it would make more sense to drop this change and set line 606 to:

```Python
elif asize > 0:
```

And add the following to the end of the block:

```Python
else:
    data = np.array([], dtype=dtype_)
```